server/controllers/finance/creditors.js

```diff
@@ -104,44 +107,58 @@ function balance(creditorUuid) {
  * @description
  * returns all transactions and balances associated with the Creditor.
  */
-function getFinancialActivity(creditorUuid) {
+function getFinancialActivity(creditorUuid, currencyId) {
   const uid = db.bid(creditorUuid);
+
   const sql = `
     SELECT trans_id, BUID(entity_uuid) AS entity_uuid, description,
-      BUID(record_uuid) AS record_uuid, trans_date, debit, credit, document, balance,
-      (@cumsum := balance + @cumsum) AS cumsum
+      BUID(record_uuid) AS record_uuid, trans_date, debit*rate as debit,
+      credit*rate as credit, document, balance*rate as balance,
+      (@cumsum := balance*rate + @cumsum) AS cumsum
     FROM (
       SELECT p.trans_id, p.entity_uuid, p.description, p.record_uuid, p.trans_date,
         SUM(p.debit_equiv) AS debit, SUM(p.credit_equiv) AS credit, dm.text AS document,
-        SUM(p.credit_equiv) - SUM(p.debit_equiv) AS balance, 0 AS posted
+        SUM(p.credit_equiv) - SUM(p.debit_equiv) AS balance, IFNULL(GetExchangeRate(e.id, ?,  p.trans_date), 1) as rate,
+        0 AS posted
       FROM posting_journal AS p
         LEFT JOIN document_map AS dm ON dm.uuid = p.record_uuid
+        
+        JOIN project pr ON pr.id = p.project_id
+        JOIN enterprise e ON e.id = pr.enterprise_id
       WHERE p.entity_uuid = ?
       GROUP BY p.record_uuid
 
       UNION ALL
 
       SELECT g.trans_id, g.entity_uuid, g.description, g.record_uuid, g.trans_date,
         SUM(g.debit_equiv) AS debit, SUM(g.credit_equiv) AS credit, dm.text AS document,
-        SUM(g.credit_equiv) - SUM(g.debit_equiv) AS balance, 1 AS posted
+        SUM(g.credit_equiv) - SUM(g.debit_equiv) AS balance, 
+        IFNULL(GetExchangeRate(e.id, ?,  g.trans_date), 1) as rate, 1 AS posted
       FROM general_ledger AS g
         LEFT JOIN document_map AS dm ON dm.uuid = g.record_uuid
+        
+        JOIN project pr ON pr.id = g.project_id
+        JOIN enterprise e ON e.id = pr.enterprise_id
+
       WHERE g.entity_uuid = ?
       GROUP BY g.record_uuid
     )c, (SELECT @cumsum := 0)z
     ORDER BY trans_date ASC, trans_id;
   `;
 
   return q.all([
-    db.exec(sql, [uid, uid]),
-    balance(creditorUuid),
```

Comment:
What happened to `balance`? 

It seems like we might have rewritten SQL code into JS and replaced `balance()` with `_.sumBy()`?  If so, that is the opposite of what we want to do.  JS struggles with math, so let's make whatever changes necessary to keep the sums in SQL.